File: programming_assignment_1/lcm.rb
def lcm(a, b)
  product = a * b
  return product / gcd(a, b)
end

def gcd(a, b)
  while b > 0 do
    remainder = a % b
    a = b
    b = remainder
  end
  a
end

if __FILE__ == $0
  a, b = gets.split().map(&:to_i)
  puts "#{lcm(a, b)}"
end

File: programming_assignment_1/lcm.py
import sys

def lcm(a, b):
    product = a * b 
    gdc = get_gcd(a, b)
    return product / gdc

def get_gcd(a, b):
    while b > 0:
        remainder = a % b
        a = b
        b = remainder
    return a

if __name__ == '__main__':
    input = sys.stdin.read()
    a, b = map(int, input.split())
    print(lcm(a, b))


